python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -459,13 +459,14 @@ def calculate_node_resources(
 
         def add_node(node_type, available_resources=None):
             if node_type not in self.node_types:
-                logger.warn(
+                logger.error(
```

Comment:
This branch of code really shouldn't be accessed, hence warn -> error. Conservatively, not throwing exception for now, though we could.

Reply:
Interesting. When this could happen?

Reply:
This could happen in current master if you change the `head_node_type` field -- that's how I found this warning.

Reply:
Might be worth adding a comment :)

Reply:
Yes, added.